### ISSUE TITLE
Fail to generate h5 file from the scripts

### ISSUE THREAD

**Issue by legend-L24** [ACEsuit/mace#1082]
I recently fine-tune MACE for water adsorption within MOFs. I try to follow the tutorial in mace GitHub to convert my large dataset to h5 format but fails. 

I use this command to convert:

#!/usr/bin/bash
mkdir processed_data
python /users/lyutao/miniforge3/envs/mace12/lib/python3.12/site-packages/mace/cli/preprocess_data.py \
    --train_file="/capstor/scratch/cscs/lyutao/cp2kdata_stress/training_data/[train.xyz](http://train.xyz/)" \
    --valid_file="/capstor/scratch/cscs/lyutao/cp2kdata_stress/training_data/[test.xyz](http://test.xyz/)" \
    --test_file="/capstor/scratch/cscs/lyutao/cp2kdata_stress/training_data/[test.xyz](http://test.xyz/)" \
    --energy_key="REF_energy" \
    --forces_key="REF_forces" \
    --atomic_numbers="[1, 6, 7, 8, 13]" \
    --r_max=5 \
    --h5_prefix="processed_data/" \
    --compute_statistics \
    --E0s='{1:-12.6261565, 6: -146.3743668, 7: -265.928468,  8: -431.60304518, 13: -52.8830503}' \
    --seed=1234 \

I get this output below and I think it should be converted successfully.

(mace12) [daint][lyutao@daint-ln003 modeldevi0]$ bash ../process.sh 
2025-07-22 22:38:02 INFO     Training file 1/1 [147463 configs, 147463 energy, 147463 forces, 147463 stresses] loaded from '/capstor/scratch/cscs/lyutao/cp2kdata_stress/training_data/[train.xyz](http://train.xyz/)'
2025-07-22 22:38:02 INFO     Total training set [147463 configs, 147463 energy, 147463 forces, 147463 stresses]
2025-07-22 22:38:14 INFO     Validation file 1/1 [16431 configs, 16431 energy, 16431 forces, 16431 stresses] loaded from '/capstor/scratch/cscs/lyutao/cp2kdata_stress/training_data/[test.xyz](http://test.xyz/)'
2025-07-22 22:38:14 INFO     Total validation set [16431 configs, 16431 energy, 16431 forces, 16431 stresses]
2025-07-22 22:38:25 INFO     Test file 1/1 [16431 configs, 16431 energy, 16431 forces, 16431 stresses] loaded from '/capstor/scratch/cscs/lyutao/cp2kdata_stress/training_data/[test.xyz](http://test.xyz/)'
2025-07-22 22:38:25 INFO     Total test set (16431 configs):
2025-07-22 22:38:25 INFO     gas_Default: 170 configs, 170 energy, 170 forces, 170 stresses
2025-07-22 22:38:25 INFO     framework+H2O_Default: 12775 configs, 12775 energy, 12775 forces, 12775 stresses
2025-07-22 22:38:25 INFO     framework_Default: 3486 configs, 3486 energy, 3486 forces, 3486 stresses
2025-07-22 22:38:25 INFO     Using atomic numbers from command line argument
2025-07-22 22:38:25 INFO     Preparing training set
2025-07-22 22:38:33 INFO     Computing statistics
2025-07-22 22:38:33 INFO     Isolated Atomic Energies (E0s) not in training file, using command line argument
2025-07-22 22:38:33 INFO     Atomic Energies: [-12.6261565, -146.3743668, -265.928468, -431.60304518, -52.8830503]
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 72/72 [00:00<00:00, 328607.06it/s]
2025-07-22 22:40:38 INFO     Average number of neighbors: 36.42812331517538
2025-07-22 22:40:38 INFO     Mean: -6.006977081298828
2025-07-22 22:40:38 INFO     Standard deviation: 1.1082674264907837
2025-07-22 22:40:38 INFO     Preparing validation set
2025-07-22 22:40:40 INFO     Preparing test sets

And I got the processed_data, which does not contain the train.h5, valid.h5. The folder only contains train, valid, and test folder, which contains a series of *h5 file. However, the mace does not accept train_dir and valid_dir as the input. My training xyz file also has config_type property. I can not generate train.h5 and val.h5 shown in the example.

Best wishes!

Yutao Li
